src/structurizr/model/element.py

```diff
@@ -82,6 +82,14 @@ def __repr__(self):
         """Return a string representation of this instance."""
         return f"{type(self).__name__}(id={self.id}, name={self.name})"
 
+    @property
```

Comment:
Should this be

```suggestion
    @abc.abstractproperty
```
if it has to be overridden by subclasses?

Reply:
I wasn't aware of that one.  Will change, thanks!

Reply:
Actually, `abstractproperty` has been deprecated since 3.3 (according to [this page](https://docs.python.org/3/library/abc.html)), in favour of using `abstractmethod` instead.  I also added a mixin to make it easier for subtypes which have no children.